ivadomed/evaluation.py

```diff
@@ -46,7 +47,12 @@ def evaluate(bids_path, log_directory, target_suffix, eval_params):
         fname_pred = os.path.join(path_preds, subj_acq + '_pred.nii.gz')
         fname_gt = [os.path.join(bids_path, 'derivatives', 'labels', subj, 'anat', subj_acq + suffix + '.nii.gz')
                     for suffix in target_suffix]
+        fname_uncertainty = os.path.join(path_preds, subj_acq + '_soft.nii.gz')
```

Comment:
I think we should be a bit clearer here as this file is not technically the uncertainty measure but rather the non-binarized average of the MC predictions.
An alernative would be to pass as parameter the suffix (eg here "soft") to control this --> eg I want to do some post processing using the CV metric --> I will have in my config file:
```
         "postprocessing": {"thr": 0.1, "suffix": "_unc-cv.nii.gz"}
```
It can also be in another PR... I am just putting this on the table.

Reply:
But .. if we go that route.. it would be a bit different from here as we will use the unc file to remove some voxels on the predictions (so we will have two files on board) etc.

Reply:
Throwing other thoughts: that could be quite time consuming.. but we could also imagine a post processing where the unc_thr is optimized on the train+valid (as done for the vox_thr) then used to perform some post-processing. (which would require to run the MC on the entire Train+Valid then perform the unc_thr optimization. Another question would be: do we optimise the vox_thr or the unc_thr first? or both simultaneously? ...etc if so how? eheh Let's brainstorm about it tomorrow!

Reply:
> I think we should be a bit clearer here as this file is not technically the uncertainty measure but rather the non-binarized average of the MC predictions.
> An alernative would be to pass as parameter the suffix (eg here "soft") to control this --> eg I want to do some post processing using the CV metric --> I will have in my config file:
> 
> ```
>          "postprocessing": {"thr": 0.1, "suffix": "_unc-cv.nii.gz"}
> ```
> 
> It can also be in another PR... I am just putting this on the table.

Yes, I like the idea, I'll do the change.



> Throwing other thoughts: that could be quite time consuming.. but we could also imagine a post processing where the unc_thr is optimized on the train+valid (as done for the vox_thr) then used to perform some post-processing. (which would require to run the MC on the entire Train+Valid then perform the unc_thr optimization. Another question would be: do we optimise the vox_thr or the unc_thr first? or both simultaneously? ...etc if so how? eheh Let's brainstorm about it tomorrow!

For this part, I'd do it in another PR since it requires a lot of changes (related to pipeline idea presented here #439) . Also currently (as you mentioned) running 10 MC iterations is extremely long even only on the test set. If we want this to be viable on train+validation set, we need to concentrate on time optimization which might require some refactoring.